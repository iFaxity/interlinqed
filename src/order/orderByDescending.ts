import { createOrderedEnumerable, createDescendingComparer, OrderedEnumerable, Enumerable, Pipe, ComparisonKey, ComparisonKeyTypes } from '../core';

/**
 * Sorts the elements of a sequence in ascending order according to key/keys.
 * @param key - A function to extract a key from an element.
 * @returns An array whose elements are sorted according the key/keys.
 */
export function orderByDescending<T, TKey extends ComparisonKeyTypes>(key: ComparisonKey<T, TKey>): Pipe<Enumerable<T>, OrderedEnumerable<T>> {
  return createOrderedEnumerable(createDescendingComparer(key));
}
