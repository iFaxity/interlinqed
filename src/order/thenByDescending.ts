import { Key, OrderedEnumerable, Pipe, createOrderedExtension, createDescendingComparer } from '../core';

/**
 * Performs a subsequent ordering of the elements in a sequence in ascending order, according to a key.
 * @param key - A function to extract a key from each element.
 * @returns A Comparison function to use with OrderBy or OrderByDescending.
 */
export function thenByDescending<T, TKey = any>(key: Key<T, TKey>): Pipe<OrderedEnumerable<T>, OrderedEnumerable<T>> {
  return createOrderedExtension(createDescendingComparer(key));
}
