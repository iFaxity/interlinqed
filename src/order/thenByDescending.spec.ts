import { describe, it, expect } from 'vitest';
import { orderByDescending } from './orderByDescending';
import { thenByDescending } from './thenByDescending';

const list = [
  { name: 'Frozen yogurt', fat: 6, carbs: 24, prot: 4 },
  { name: 'Ice cream sandwich', fat: 9, carbs: 37, prot: 4.3 },
  { name: 'Eclair', fat: 16, carbs: 24, prot: 6 },
  { name: 'Gingerbread', fat: 16, carbs: 49, prot: 0 },
  { name: 'Jelly bean', fat: 0, carbs: 94, prot: 0 },
];

describe('#thenByDescending()', () => {
  it('test1', () => {
    const res = [ ...thenByDescending<any>(x => x.fat)(orderByDescending<any>(x => x.carbs)(list)) ];

    expect(res).toMatchSnapshot();
  });

  it('test2', () => {
    expect(() => thenByDescending<any>(x => x.fat)(list as any)).toThrow();
  });
});
