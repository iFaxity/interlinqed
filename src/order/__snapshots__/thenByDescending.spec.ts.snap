// Vitest Snapshot v1

exports[`#thenByDescending() > test1 1`] = `
[
  {
    "carbs": 94,
    "fat": 0,
    "name": "Jelly bean",
    "prot": 0,
  },
  {
    "carbs": 49,
    "fat": 16,
    "name": "Gingerbread",
    "prot": 0,
  },
  {
    "carbs": 37,
    "fat": 9,
    "name": "Ice cream sandwich",
    "prot": 4.3,
  },
  {
    "carbs": 24,
    "fat": 16,
    "name": "Eclair",
    "prot": 6,
  },
  {
    "carbs": 24,
    "fat": 6,
    "name": "Frozen yogurt",
    "prot": 4,
  },
]
`;
